sync/sync.go

```diff
@@ -231,6 +282,12 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 			s.newHeads.Send(block.Header)
 			s.log.Infow("Stored Block", "number", block.Number, "hash",
 				block.Hash.ShortString(), "root", block.GlobalStateRoot.ShortString())
+			if s.plugin != nil {
```

Comment:
flow is a little bit different compared if block reverted: check `s.plugin != nil` happens inside `verifierTask`,  but for revert it's inside `handlePluginRevertBlock`. I don't have preferences on the way it should be handled, either in separate functions or in `verifierTask`

Reply:
Also for new block: we first store it and then notify plugin.
For revertion: we first notify plugin and then revert it in db

Reply:
> For revertion: we first notify plugin and then revert it in db

Yup, I wanted to ping the plugin after juno reverts, but this turned out to be a bit messy. The reason is that after Juno reverts, we no longer have the block and state-update that just got reverted. I could collect these before juno reverts, and then pass them into the logic that pings the plugin, but this felt like it would be messy. So I just decided to ping the plugin first.

Reply:
>flow is a little bit different compared if block reverted

Yup, I'll update the code to be more consistent (handle the nil inside the plugin handler)

Reply:
>flow is a little bit different compared if block reverted: check s.plugin != nil happens inside verifierTask, but for revert it's inside handlePluginRevertBlock

Updated to be more consistent (move nil handle outside of revert handler, since SendBlock() doesn't have a handler to put it in)